exercises/custom-set/example.exs

```diff
@@ -1,136 +1,74 @@
 defmodule CustomSet do
-  # This lets the compiler check that all Set callback functions have been
-  # implemented.
-  @behaviour Set
+  defstruct map: Map.new
 
-  # Use a list as set implementation. This is horribly slow compared to other
-  # approaches, but simple.
+  @opaque t :: %__MODULE__{map: map}
 
-  # Wrapper for the internal list
-  defstruct list: []
-
-  def new() do
-    %CustomSet{}
-  end
-
-  def new(coll) do
-    Enum.reduce(coll, %CustomSet{}, fn x, s -> put(s, x) end)
-  end
-
-  ## Set callbacks
-  def delete(s, x) do
-    wrap(List.delete(unwrap(s), x))
-  end
-
-  def difference(a, b) do
-    wrap(Enum.filter_map(diff(a, b), &(match?({:a_only, _}, &1)), &(elem(&1, 1))))
-  end
-
-  def disjoint?(a, b) do
-    not Enum.any?(diff(a, b), &(match?({:both, _}, &1)))
-  end
-
-  def empty(_) do
-    new()
-  end
-
-  def equal?(a, b) do
-    unwrap(a) === unwrap(b)
+  @spec new(Enum.t) :: t
+  def new(enumerable) do
+    %CustomSet{map: Map.new(enumerable, fn (x) -> {x, true} end)}
```

Comment:
I didn't see a test where this behavior of `new` is demonstrated. Perhaps we need a set of tests showing that `[1,1,2,3]` creates a CustomSet of `[1,2,3]`?